### ISSUE TITLE
Misleading CS0123 error about delegates to ref structs

### ISSUE THREAD

**Issue by Happypig375** [dotnet/roslyn#29737]
**Version Used**: 
Visual Studio 15.8.3 integrated C# Interactive

**Steps to Reproduce**:

Execute this in the C# Interactive
```cs
ref struct A { public int M() => 0; }
public Func<int> Test() { var a = new A(); return (Func<int>)a.M; }
```

**Expected Behavior**:
The compiler error tell me about not allowing construction of a delegate to a ref struct

**Actual Behavior**:
error CS0123: No overload for 'M' matches delegate 'Func<int>'

but the method is clearly there.

**PS**: Would the proposed static delegates allow pointing to methods of ref structs?
